File: outfit-planner-system/outfit-service/src/main/java/com/outfit/planner/system/outfit/service/business/OutfitsService.java
package com.outfit.planner.system.outfit.service.business;

import com.outfit.planner.system.outfit.service.business.model.CreateOutfitRequest;
import com.outfit.planner.system.outfit.service.business.model.PredictOutfitsRequest;
import com.outfit.planner.system.outfit.service.business.model.ToggleWearOutfitRequest;
import com.outfit.planner.system.outfit.service.dto.OutfitDto;

import java.util.List;

public interface OutfitsService {

    OutfitDto createOutfit(CreateOutfitRequest request);
    List<OutfitDto> predictOutfits(PredictOutfitsRequest username);
    boolean toggleWorn(ToggleWearOutfitRequest request);
}

File: outfit-planner-mf/libs/shared/components/src/lib/services/outfits.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable, shareReplay, tap} from "rxjs";
import {Cache, InvalidateCache, SharedCacheService} from "./shared-cache.service";
import {CreateOutfitRequest, Outfit, PredictOutfitsRequest, ToggleWearOutfitRequest} from "../defs";

@Injectable({
  providedIn: 'root'
})
export class OutfitsService{

  constructor(private httpClient: HttpClient) {
  }

  @Cache("outfits")
  predictOutfits(request: PredictOutfitsRequest): Observable<Outfit[]> {
    return this.httpClient.post<Outfit[]>('/api/outfits/predictions', request);
  }

  @InvalidateCache("outfits")
  addOutfit(request: CreateOutfitRequest): Observable<Outfit> {
    return this.httpClient.post<Outfit>('/api/outfits', request);
  }

  @InvalidateCache("outfits")
  toggleWear(request: ToggleWearOutfitRequest): Observable<boolean> {
    return this.httpClient.post<boolean>('/api/outfits/toggle-wear', request);
  }

}
